sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/PartitionLoadBalancerImpl.java

```diff
@@ -108,7 +108,9 @@ public boolean isRunning() {
             .flatMap(allLeases -> {
                 if (cancellationToken.isCancellationRequested()) return Mono.empty();
                 List<Lease> leasesToTake = this.partitionLoadBalancingStrategy.selectLeasesToTake(allLeases);
-                this.logger.debug("Found {} leases, taking {}", allLeases.size(), leasesToTake.size());
+                if (leasesToTake.size() > 0) {
+                    this.logger.info("Found {} total leases, taking ownership of {}", allLeases.size(), leasesToTake.size());
```

Comment:
I think it will be helpful to always log even though there is no lease to take

Reply:
actually no, because it will introduce more logs that we don't actually need. We know that this loop is executed, it's when leases are picked up either because they are expired or the equal load balancing is not achieved, that it's a problem.